### ISSUE TITLE
feat: Implement type imports proposal

### ISSUE THREAD

**Issue by hydai** [WasmEdge/WasmEdge#1124]
## Introduction

This is one of the requirements for GC proposal.

[Type imports](https://github.com/WebAssembly/proposal-type-imports) proposal allows Wasm modules to import type definitions.

However, this proposal depends on #1123. Details: https://github.com/WebAssembly/proposal-type-imports/blob/master/proposals/type-imports/Overview.md#proposal-summary

## Motivation
 
TBD.

## Details

TBD.

## Appendix

1. GC Proposal: https://github.com/WebAssembly/gc
2. Typed function references Proposal: https://github.com/WebAssembly/function-references

**Comment by q82419** [comment]
Merged into the GC/typed-func-ref proposal and finished.
